### DISCUSSION TITLE
Cannot dumpeventtexts

### DISCUSSION THREAD

**Discussion by lIllIIlI** [FakeFishGames/Barotrauma#14342] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

Trying to translate some chinese mods to english. i.e Touhou Deep Sea Operation
Instead I get this:
![image](https://github.com/user-attachments/assets/1d6ff809-783b-4edf-8f98-c26d77858b72)

### Reproduction steps

Subscribe to this mod.
https://steamcommunity.com/sharedfiles/filedetails/?id=2941124988&searchtext=touhou
Do dumpeventtexts in debug console
voila. red text

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Other

### -

Main menu

### Version

v1.5.9.1 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

```shell
Cannot write all lines to "Content/Texts/EventTexts.txt": modifying the files in this folder/with this extension is not allowed.Failed to open the file "Content/Texts/EventTexts.txt". {An error occurred trying to start process 'A:\SteamLibrary\steamapps\common\Barotrauma\Content\Texts\EventTexts.txt' with working directory 'A:\SteamLibrary\steamapps\common\Barotrauma'. The system cannot find the file specified.}
   at System.Diagnostics.Process.StartWithShellExecuteEx(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start()
   at System.Diagnostics.Process.Start(ProcessStartInfo startInfo)
   at Barotrauma.ToolBox.OpenFileWithShell(String filename) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Utils\ToolBox.cs:line 507
   at Barotrauma.DebugConsole.<>c.<InitProjectSpecific>b__23_99(String[] args) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\DebugConsole.cs:line 2779
Cannot write XML document to "A:\SteamLibrary\steamapps\common\Barotrauma\Content\TraitorEvents.xml": modifying the files in this folder/with this extension is not allowed.
Cannot write to invalid XmlWriter
Cannot flush invalid XmlWriter
Cannot dispose invalid XmlWriter
```

**Comment by lIllIIlI** [comment]
Actually, it seems even if you are not using any mods, the same thing happens.

**Comment by ubertpendragon** [comment]
It's broken in several ways, and honestly not intuitive in its description

Here's what I did:


Step 1: 
Create a SEPARATE mod with the event file using written dialog in it

Step 2: 
Created a Eventtext.xml file already pre-formatted
![image](https://github.com/user-attachments/assets/4014db76-c86c-4cc5-a514-0e87ea64b25c)

Step 3:
Typed in console
`dumpeventtexts G:/SteamLibrary/steamapps/common/Barotrauma/LocalMods/Novevents/eventtext.xml`

Step 4:
Still a shitton of errors, the command is probably trying to dump vanilla event text???
![image](https://github.com/user-attachments/assets/fa806b25-8934-41ff-962b-24be1183a9b0)

Step 5: Get parasite lorem ipsum text from out of untranslated vanilla files (I couldnt find those text identifier in vanilla files)
![image](https://github.com/user-attachments/assets/29dd7f04-4429-464c-af50-b68f941e033f)

**Comment by ubertpendragon** [comment]
I had to write a guide on our discord to use this command

# GUIDE ON HOW TO USE THE DUMPEVENTTEXT COMMAND (HIGHLY BROKEN):

**STEP 1: CREATE A SEPARATE MOD**
Add the following files:
* filelist.xml
* event.xml <= your file with events
* eventtext.xml <= a pre-formatted, empty text file (see above)
NAME THOSE FILES HOWEVER YOU WANT

**STEP 2: **
OPEN BAROTRAUMA, LOAD ONLY THIS NEW MOD, REMOVE OTHER MODS

**STEP 3: **
OPEN CONSOLE, TYPE `G:/SteamLibrary/steamapps/common/Barotrauma/LocalMods/Novevents/EventText.xml`

**STEP 4: **
TEXT HAS GENERATED IN EVENTTEXT.XML, AND RAW TEXT HAS BEEN CONVERTED IN EVENT.XML

**STEP 5:**
THE COMMAND WILL TRY TO ALSO CONVERT VANILLA TEXT, LAG YOUR GAME AND SEND A SHIT TON OF MESSAGES. IGNORE IT

**STEP 6: **
COPY THOSE FILES AND REPLACE THE FILES IN THE MAIN MOD. IF YOU'VE CREATED A NEW TEXT FILE, DON'T FORGET TO ADD IT TO FILELIST

**STEP 7: **
PUT THE ELEMENTS BACK INTO THE TEXT FILE, THEY GET DELETED FOR SOME REASON (STUPID AF)

`<?xml version="1.0" encoding="utf-8"?>
<infotext> 
</infotext>`

**Comment by Regalis11** [comment]
Thank you for the report! We actually only added this command for internal use (back when we needed to move a large number of event texts to a localization file) and didn't expect it to be used by modders, which is why it tries to modify the vanilla files. I can see how the command would be useful in mod development too though, I think we can look into making it a little easy to use on non-vanilla content.

**Comment by Jasontti** [comment]
Tested a fix for this and dumpeventtexts can be used on a single file or a folder containing events.
For single files:
`dumpeventtexts LocalMods/TestEventTexts/TestEvent.xml LocalMods/EventText.txt`
For a folder
`dumpeventtexts LocalMods/TestEventTexts/Events LocalMods/EventText.txt`

Mind you that dumping a folder doesn't work in the root of localmod, recommended to put them in a separate folder.
Closing discussion as resolved.
